src/main/java/org/lmdbjava/Dbi.java

```diff
@@ -186,12 +190,28 @@ public boolean delete(final Txn<T> txn, final T key, final T val) {
    * @param txn transaction handle (not null; not committed; must be R-W)
    */
   public void drop(final Txn<T> txn) {
+    drop(txn, false);
+  }
+
+  /**
+   * Drops the database. If delete is set to true, the database will be deleted
+   * and handle will be closed. See {@link #close()} for implication of handle close.
+   * Otherwise, only the data in this database will be dropped.
+   *
+   * @param txn transaction handle (not null; not committed; must be R-W)
+   * @param delete whether database should be deleted.
+   */
+  public void drop(final Txn<T> txn, final boolean delete) {
     if (SHOULD_CHECK) {
       requireNonNull(txn);
       txn.checkReady();
       txn.checkWritesAllowed();
     }
-    checkRc(LIB.mdb_drop(txn.pointer(), ptr, 0));
+    if (delete) {
+      clean();
+    }
+    final int del = delete ? 1 : 0;
+    checkRc(LIB.mdb_drop(txn.pointer(), ptr, del));
   }
```

Comment:
Make close() idempotent is also a viable approach since lmdb doc also suggest only closing a dbi once. http://www.lmdb.tech/doc/group__mdb.html#gac08cad5b096925642ca359a6d6f0562a